webpack.config.js

```diff
@@ -0,0 +1,103 @@
+const path = require('path');
+const webpack = require('webpack');
+const CleanWebpackPlugin = require('clean-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const ExtractTextWebpackPlugin = require('extract-text-webpack-plugin');
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+
+var minify = {};
+
+if (process.env.NODE_ENV === 'production') {
+    minify = {
+        collapseInlineTagWhitespace: true,
+        collapseWhitespace: true
+    };
+}
+
+module.exports = {
+    entry: './src/js/main.js',
+    output: {
+        path: path.resolve(__dirname, './dist'),
+        publicPath: '/',
+        filename: '[name].[hash].js'
+    },
+    module: {
+        rules: [
+            {
+                enforce: 'pre',
+                test: /\.vue$/,
+                loader: 'eslint-loader',
+                exclude: /node_modules/
+            },
+            {
+                test: /\.vue$/,
+                loader: 'vue-loader',
+                options: {
+                    extractCSS: true
```

Comment:
I know we did it together but maybe CSS extraction should depends on build mode (dev or prod)